pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -1052,7 +1052,15 @@ func (cb *ClusterBuilder) buildUpstreamClusterTLSContext(opts *buildClusterOpts,
 		if opts.serviceRegistry == provider.External && len(tls.SubjectAltNames) == 0 {
 			tls.SubjectAltNames = opts.serviceAccounts
 		}
-		if tls.CredentialName != "" {
+		if tls.InsecureSkipVerify != nil && tls.InsecureSkipVerify.Value {
+			tls.CaCertificates = ""
+			tls.SubjectAltNames = []string{}
```

Comment:
why do you change the input tls settings

Reply:
This is for testing but I am thinking it will be best to remove this and make an integration test to replace the unit test